File: MyApplicationmm/app/src/main/java/com/example/myapplicationmm/MainActivity.java
package com.example.myapplicationmm;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
private ArrayList<String> d;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ArrayList<String> ss = new ArrayList<>();
        foo(ss);

    }
    private void foo(ArrayList<String> d1){
        this.d=d1;
        Log.d("before",String.valueOf(this.d==d1) );
        d1= new ArrayList<>();
        Log.d("before",String.valueOf(this.d==d1) );
    }
}

File: TestSearch/app/src/main/java/com/example/testsearch/MainActivity.kt
package com.example.testsearch

import android.app.SearchManager
import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.ContextMenu
import android.view.Menu
import android.view.MenuInflater
import android.view.View
import android.content.ComponentName
import android.util.Log
import androidx.appcompat.widget.SearchView


class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        Log.d("test","Ol");
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        val inflater : MenuInflater = menuInflater
        inflater.inflate(R.menu.menu, menu)
        val searchManager : SearchManager  = getSystemService(Context.SEARCH_SERVICE) as SearchManager
        val searchView : SearchView = menu?.findItem(R.id.search_View)?.actionView as SearchView
        searchView.setSearchableInfo(searchManager.getSearchableInfo(ComponentName(this, SearchActivity::class.java)))
        searchView.setIconifiedByDefault(false);
        return true
    }
}